src/blockchain/shelley/utils.js

```diff
@@ -0,0 +1,180 @@
+          pool_owners.push(keyBytes.toString('hex'))
+        }
+        common.certificate = {
+          type: 'PoolRegistration',
+          pool_id: reg.id().to_string(),
+          // we should be able to do this considering js max int would be 28,5616,414 years
+          start_validity: parseInt(reg.start_validity().to_string(), 10),
+          owners: pool_owners,
+        }
+        break
+      }
+      case wasm.CertificateType.StakeDelegation: {
+        const deleg = cert.get_stake_delegation()
+        const poolId = deleg.delegation_type().get_full()
+        common.certificate = {
+          type: 'StakeDelegation',
+          // TODO: possibly handle Ratio types
+          pool_id: poolId != null ? poolId.to_string() : null,
+          account: deleg.account().to_hex(),
+        }
+        break
+      }
+      case wasm.CertificateType.PoolRetirement: {
+        const retire = cert.get_pool_retirement()
+        common.certificate = {
+          type: 'PoolRetirement',
+          pool_id: retire.pool_id().to_string(),
+          // we should be able to do this considering js max int would be 28,5616,414 years
+          retirement_time: parseInt(retire.retirement_time().to_string(), 10),
+        }
+        break
+      }
+      case wasm.CertificateType.PoolUpdate:
+        console.log('\n\n\n\n\n========\n\nPOOL UPDATE FOUND\n\n\n')
+        break
+      case wasm.CertificateType.OwnerStakeDelegation: {
+        if (inputs_parsed.length != 1 || inputs_parsed[0].type != 'account') {
+          throw new Error(`Malformed OwnerStakeDelegation. Expected 1 account input, found: ${JSON.stringify(inputs_parsed)}`)
+        }
+        const deleg = cert.get_owner_stake_delegation()
+        const poolId = deleg.delegation_type().get_full()
+        common.certificate = {
+          type: 'OwnerStakeDelegation',
+          // TODO: possibly handle Ratio types
+          pool_id: poolId != null ? poolId.to_string() : null,
+          account: inputs_parsed[0].account_id,
+        }
+        break
+      }
+      default:
+        throw new Error(`parsing certificate type not implemented: ${cert.get_type()}`)
+    }
+  }
+  const ret = {
+    inputs: inputs_parsed,
+    outputs: outputs_parsed,
+    witnesses: [],
+    ...common,
+    ...extraData,
+  }
+  console.log(`parsed a tx: \n${JSON.stringify(ret)}\n`)
+  return ret
+  // const [[inputs, outputs], witnesses] = tx
+  // const [txId, txBody] = packRawTxIdAndBody(tx)
+  // return {
+  //   id: txId,
+  //   inputs: inputs.map(inp => {
+  //     const [type, tagged] = inp
+  //     const [inputTxId, idx] = cbor.decode(tagged.value)
+  //     return { type, txId: inputTxId.toString('hex'), idx }
+  //   }),
+  //   outputs: outputs.map(out => {
+  //     const [address, value] = out
+  //     return { address: bs58.encode(cbor.encode(address)), value }
+  //   }),
+  //   txBody,
+  //   ...extraData,
+  // }
+}
+
+const rawTxToObj = (tx: Array<any>): TxType => {
```

Comment:
This needs to be changed. I never tested it and it won't work since we pass in a Block....

Sebastien just added in a `Fragment::from_bytes()` in Emurgo/js-chain-libs#20 so we can just swap that out and hope it works (don't see why not we don't do so many checks like with byron). We'd need to write a `Validator` for Shelley though first since due to how the inversify lib works I think we need one and null won't work.

Reply:
Updated js-chain-libs and used the proposed function